src/python/pants/option/options_test.py

```diff
@@ -1733,3 +1748,82 @@ class DummyOptionable1(Optionable):
         single_warning_dummy1 = assert_single_element(w)
         self.assertEqual(single_warning_dummy1.category, DeprecationWarning)
         self.assertEqual("vv", vals1.foo)
+
+    def test_list_of_enum_single_value(self) -> None:
+        options = self._parse(flags="other-enum-scope --some-list-enum=another-value")
+        assert [self.SomeEnumOption.another_value] == options.for_scope(
+            "other-enum-scope"
+        ).some_list_enum
+
+    def test_list_of_enum_default_value(self) -> None:
+        options = self._parse(flags="other-enum-scope --some-list-enum-with-default=another-value")
+        assert [
+            self.SomeEnumOption.yet_another,
+            self.SomeEnumOption.another_value,
+        ] == options.for_scope("other-enum-scope").some_list_enum_with_default
+        options = self._parse()
+        assert [self.SomeEnumOption.yet_another] == options.for_scope(
+            "other-enum-scope"
+        ).some_list_enum_with_default
+
+    def test_list_of_enum_multiple_values(self) -> None:
+        options_1 = self._parse(flags="other-enum-scope --some-list-enum=another-value,one-more")
```

Comment:
This is bad behavior. Pants does not support using commas to signify multiple values with a `list` option because there is ambiguity if the string was intended to have a comma. For example:

```
./pants --coverage-py-filter="v1,v2" options | grep coverage-py.filter
coverage-py.filter = ['v1,v2'] (from FLAG)
```

The only way to specify multiple options with a `list` is by either using explicit list syntax like `['v1', 'v2']` or to pass the flag multiple times. https://pants.readme.io/docs/options#list-values

Reply:
Is there an actual use case for specifying values when a comma is a valid value?
I find putting square brackets on command line arguments counterintuitive, this is not what I expect from a CLI tool.
for now, this is only enabled for enum values which I think makes for a more intuitive:
`./v2 test --use-coverage --coverage-py-report="['xml', 'html']"` vs
`./v2 test --use-coverage --coverage-py-report=xml,html`

In this case, I don't see the downside of keeping this, especially since it is highly unlikely for enum values to have commas in them.
I think we should make things more user friendly and not require an awkward and weird syntax where we don't need to.

Reply:
@stuhood @benjyw any thoughts here?

Reply:
I think comma-separated enum values makes sense (we can ban commas from the values themselves). And it also makes sense for string-valued lists in general (strings rarely need to contain commas, so I think it's fine to only require quoted strings when they do).

Reply:
To clarify - there has to be an escape hatch for the case when strings do contain embedded commas. So we'd need to split only strings that aren't surrounded by `"`, say. Anything that is we just parse. But may be this is now best done in a followup change, so we can be sure it's properly tested?

Reply:
The behavior of this syntax would be confusing I think. Given:
```
# Clears the list and then sets it to xml, html
--coverage-py-report="['xml', 'html']"
# Adds to the default values of the list:
--coverage-py-report=xml
--coverage-py-report="+['xml']"
# Does ... one of those things probably?
--coverage-py-report=xml,html
```

Having three+ ways to do things is costly, and I don't think it's worth it here.

Reply:
With Benjy's point about an escape hatch for when you do really want the `,` included, I agree this is a nice feature to have.

However, could you please split it out into a dedicated PR? That's a non-trivial change, so it is best with a dedicated PR for proper discussion and so that we can ensure we've thought through all the details like how it works with config files.

Reply:
again, the support for *commas is for the new feature (list of enums)* only and doesn't impact any other types/existing features.
It doesn't break existing behavior and I don't think that warrants the overhead of splitting it to another PR.

Existing lists of string values, when the string values have commas, will continue to work as they were, this change doesn't impact them. since we only parse/split on comma for lists when the member type is an enum.

Reply:
>  the support for commas is for the new feature (list of enums) only and doesn't impact any other types/existing features.

Understood. But you are adding a new syntax still to the options system, and this new syntax is special cased. In general, we try to be careful to not special case things because it introduces new complexity and permutations to the project.

If we wanted to support commas with enums, I believe we would want to support it with strings too. Hence why I think this is best saved for a dedicated PR.

Reply:
> again, the support for commas is for the new feature (list of enums) only and doesn't impact any other types/existing features.

That makes it harder to document, I think. Because you'd need to explain it as a special case on https://pants.readme.io/docs/options#list-values ... It would be better for it not to be a special case, but that runs into the ambiguities above.